src/libraries/System.Private.CoreLib/src/System/SpanHelpers.cs

```diff
@@ -406,8 +420,35 @@ public static unsafe void ClearWithReferences(ref IntPtr ip, nuint pointerSizeLe
 
             // Write only element.
             ip = default;
+            return;
+
+        LargeInput:
+            // Branch-less alignment to 64 bytes.
+            Unsafe.As<nint, Block64>(ref ip) = default;
```

Comment:
This makes fragile assumption about the instruction that the JIT is going to compile this into. It is valid for the JIT to compile this into one-byte-at-time assignment.